server/src/main/scala/ch.epfl.scala.index.server/routes/Oauth2.scala

```diff
@@ -20,13 +20,13 @@ class OAuth2(github: Github, session: GithubUserSession) {
   val routes =
     get {
       path("login") {
-        headerValueByType[Referer]() { referer =>
+        optionalHeaderValueByType[Referer]() { referer =>
           redirect(
             Uri("https://github.com/login/oauth/authorize").withQuery(
               Query(
                 "client_id" -> github.clientId,
                 "scope" -> "read:org",
-                "state" -> referer.value
+                "state" -> referer.map(_.value).getOrElse("/")
```

Comment:
Will a relative URL work? I wonder if github can then figure out where to redirect back?